### PR TITLE
ci: use k8s pod template agents

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

PoC with K8s CI Workers

## Checklist

- ~~[ ] This is an enhancement of existing features, or a new feature in existing plugins~~
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/master/CHANGELOG.asciidoc)
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] I have made corresponding changes to the documentation
- ~~[ ] This is a bugfix~~
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/master/CHANGELOG.asciidoc)
  - [ ] I have added tests that would fail without this fix
- ~~[ ] This is a new plugin~~
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/master/CHANGELOG.asciidoc)
  - [ ] My code follows the [style guidelines of this project](https://github.com/elastic/apm-agent-java/blob/master/CONTRIBUTING.md#java-language-formatting-guidelines)
  - [ ] I have made corresponding changes to the documentation
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] New and existing [**unit** tests](https://github.com/elastic/apm-agent-java/blob/master/CONTRIBUTING.md#testing) pass locally with my changes
  - [ ] I have updated [supported-technologies.asciidoc](https://github.com/elastic/apm-agent-java/blob/master/docs/supported-technologies.asciidoc)
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] Added an instrumentation plugin? Describe how you made sure that old, non-supported versions are not instrumented by accident.
- [x] This is something else
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/master/CHANGELOG.asciidoc)

## Why 

1) Delegate the Agents configuration to the consumers
2) Faster provisioning.

For instance, the agent is provisioned almost immediately, compared to the 2 minutes average per CI worker

image

## Actions

- [ ] Cached workspace to speed up builds?
- [ ] Matrix support for difference openjdk versions

### CODE REVIEW

File: .ci/k8s/OpenJdkPod.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+kind: Pod
+spec:
+  containers:
+  - name: jdk11
+    image: openjdk:11-jdk-buster
+    command:
+    - sleep
+    args:
+    - infinity
+## TODO: Faster builds with some cached artifacts.
```
Comment (v1v) [inline .ci/k8s/OpenJdkPod.yml:11]: Interesting enough:

image

vs

image

**Comment by v1v** [comment]
@mergifyio refresh

File: Jenkinsfile [hunk 2]
```diff
@@ -155,24 +157,24 @@ pipeline {
           Run smoke tests for different servers and databases.
         */
         stage('Smoke Tests 01') {
-          agent { label 'linux && immutable' }
+          agent { kubernetes { yamlFile '.ci/k8s/OpenJdkPod.yml' } }
```
Comment (v1v) [inline Jenkinsfile:160]: IIRC, this was the way to ensure the parallel stages don't clash with the workspace when using the top-level k8s definition.

I need to clarify this to be sure this is the right pattern to be used.

**Comment by v1v** [comment]
@Mergifyio update

**Comment by eyalkoren** [comment]
@v1v do we want to proceed with this, or can I close this PR?

**Comment by v1v** [comment]
Closing this issue as we are moving away from Jenkins

**Pull request closed without merging**